### DISCUSSION TITLE
Build.sh

### DISCUSSION THREAD

**Discussion by tathastu871** [termux/termux-packages#8303] [Q&A] [answered]
package has SRCURL a git repo no tags or release and hence no version. so should TERMUX_PKG_VERSION be kept blank. checks were failing due to version not set

**Comment by agnostic-apollo** [comment, marked as answer]
`TERMUX_PKG_VERSION` must always be set, otherwise how would `apt`/`dpkg` ever be able to update that package? Repo may have a version defined in some version file or in code, use that. If it does not, then get it defined, or possibly use your own for now with github commit hash as url. Do not use `master` branch as url since checksum will change when new commits are pushed.

For checksum/pre-release sample, check https://github.com/termux/termux-packages/pull/8198.

**Reply by tathastu871** [reply]
Theres a package xpup. Its actually fully developed no new updates are going to be pushed. It was from same developer of package pup that was already in termux repo. Xpup can be installed using go get. it log says

go get github.com/ericchiang/xpup

go: downloading github.com/ericchiang/xpup v0.0.0-20150709172731-8b964504de6f

Based on above url should i set version to 0.0.0 ?? 


SHOULD I WRITE
TERMUX_PKG_VERSION=0.0.0-20150709172731
_COMMIT=8b964504de6fe798bfcdfb0e5303517342ceab42

?? Please answer

**Reply by agnostic-apollo** [reply]
You can check version syntax requirements for `deb` `control` files at https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-version. Since git commit hashes are not sequential/sortable, having them in version names can complicate things. Prefixing them with timestamps should allow it to work. Add `-1` add end, in case it needs to be bumped.

https://unix.stackexchange.com/questions/640237/deb-version-is-git-taggit-commit-hash-apt-downloading-the-wrong-version-fro

```
TERMUX_PKG_VERSION=0.0.0+20150709172731.8b964504de6f-1
TERMUX_PKG_SRCURL="https://github.com/ericchiang/xpup/archive/8b964504de6fe798bfcdfb0e5303517342ceab42.tar.gz"
TERMUX_PKG_SHA256=<checksum>
```
After manually downloading `tar`, find its checksum with `sha256sum` command and set it in `TERMUX_PKG_SHA256`.
